### ISSUE TITLE
options disappear when build_requires are defined in conanfile.py

### ISSUE THREAD

**Issue by ntff** [conan-io/conan#1384]
conanfile.py with only `requires` specified:

```python
from conans import ConanFile

class package(ConanFile):
    name            = "issue_test"
    requires        = "boost/1.64.0@user/stream"
    options         = {"coverage": [True, False]}
```

conan command passes successfully:
```shell
user@:~$ conan install --options coverage=True
```

conanfile with 'build_requires' specified:
```python
from conans import ConanFile

class package(ConanFile):
    name            = "issue_test"
    build_requires  = "boost/1.64.0@user/stream"
    options         = {"coverage": [True, False]}
```
with the same `conan install --options coverage=True` command
produces

```
...
ERROR: 'options.coverage' doesn't exist
Possible options are []
```

0.23.1 and 0.23.2 were tested.

**Comment by memsharded** [comment]
That is very weird, it should also fail in the first case.
Please try including the package in the command line, as:

```bash
$ conan install --options issue_test:coverage=True
```
I will have a look asap, thanks for reporting.

**Comment by ntff** [comment]
Everything is the same with package name included.

**Comment by memsharded** [comment]
Could you please provide some fully reproducible example? For example, the above snippet is missing the version field. I keep working on it, but if you can, that would be faster. Thanks!

**Comment by memsharded** [comment]
Everything seems good here, I haven't been able to reproduce the issue. Maybe showing the complete workflow, with exports, installs, etc. might help.

**Comment by ntff** [comment]
These examples are minimal working without warnings.
Only that dependency `boost/1.64.0@user/stream` is from my private repository - just put any package you have there.

**Comment by ntff** [comment]
I'll try to reproduce including fresh system install - this will take some time. Thanks.

**Comment by memsharded** [comment]
Yes, that is what I did. I added a version number, and changed the dependency to a simple header only from bintray conan-transit repo. ``conan export``, then ``conan install <pkg_ref> --build=missing -o coverage=True`` seems to work fine for both build_requires and requires. Any idea what could I be missing?

**Comment by ntff** [comment]
I do in-workspace install/build, without exporting to local repository.

Build after exporting there works, but it's tedious to export each time so, I just install and build.
And it worked!

**Comment by memsharded** [comment]
Oh, I see, it is for consuming! Sorry I didn't check that case. Yes, I can confirm this behavior, it is weird. Label as bug, it should have priority, I will assign 0.24, though it might be moved to 0.25, because 0.24 is almost already closed. Lets see what we can do. Thanks for reporting!

**Comment by memsharded** [comment]
Fixed, indeed a subtle bug with the application of options to both normal dependencies and build_requires. Submitted PR #1387, should be in next release. Thanks for helping to fix it.

**Comment by memsharded** [comment]
This has been fixed and released in 0.24, could you please try it and report? Thanks!
